pkg/model/names.go

```diff
@@ -116,14 +116,21 @@ func (b *KopsModelContext) LinkToIAMInstanceProfile(ig *kops.InstanceGroup) *aws
 	return &awstasks.IAMInstanceProfile{Name: &name}
 }
 
-// SSHKeyName computes a unique SSH key name, combining the cluster name and the SSH public key fingerprint
+// SSHKeyName computes a unique SSH key name, combining the cluster name and the SSH public key fingerprint.
+// If an SSH key name is provided in the cluster configuration, it will use that instead.
 func (c *KopsModelContext) SSHKeyName() (string, error) {
```

Comment:
> when we do a delete a cluster do we delete the key?

I don't think we should. I'm not sure what's been done in the past, but I think that things created by kops should be deleted by kops and things not created by kops should not be deleted by kops.

I don't fully understand the current behavior on deletion (the task lifecycle is hard to follow in my IDE), but my intention is not to delete.